src/cache_cf.cc

```diff
@@ -633,6 +634,75 @@ parseConfigFile(const char *file_name)
     }
 }
 
+/*
+ * The templated functions below are essentially ConfigParser methods. They are
+ * not implemented as such because our generated code calling them is the only
+ * code that can instantiate implementations for each T -- we cannot place these
+ * definitions into ConfigParser.cc unless cf_parser.cci is moved there.
+ */
+
+// TODO: When adding Ts incompatible with this trivial API and implementation,
+// replace both with a ConfigParser-maintained table of seen directives.
+/// whether we have seen (and, hence, configured) the given directive
+template <typename T>
+static bool
+SawDirective(const T &raw)
+{
+    return bool(raw);
+}
+
+/// Sets the given raw SquidConfig data member.
+/// Extracts and interprets parser's configuration tokens.
+template <typename T>
+static void
+ParseDirective(T &raw, ConfigParser &parser)
+{
+    if (SawDirective(raw))
```

Comment:
We have quite a few directives that can be repeated over multiple lines. For this template to replace the legacy API it needs to support those. Perhapse shift this check into Config::Component<T> instead?

Reply:
> We have quite a few directives that can be repeated over multiple lines. For this template to replace the legacy API it needs to support those

Yes, we will support more complex duplicate rejection logic. The SawDirective() TODO acknowledges that future work. However, we should not add that support in this PR because such support is not needed by the new API users, because proper support is difficult to design, and because designing APIs without specific users often leads to poor results.

> Perhapse shift this check into Config::Component instead?

We will make that and other complex decisions when we have to. Today, Config::Component embodies `TYPE:`-driven logic. Shifting this check into Config::Component today would imply that `TYPE` determines whether a particular directive can be repeated. That implication may be false. I would not be surprised if future duplicate handling will be driven by code _generated_ from some new cf.data.pre metadata other than `TYPE`!

I am sure ParseDirective() will change from its current simple form as we add support for more configuration cases. There is nothing wrong with that anticipated change as long as the current code addresses our current needs (and no simple changes can make it better). Trying to predict what other cases will need is not going to work well here because that decision space is very complex. Let's keep it simple while we can.